core/src/main/java/hudson/cli/CLICommand.java

```diff
@@ -642,4 +653,15 @@ public static CLICommand getCurrent() {
             }
         }
     }
+
+    /**
+     * User details loaded from the CLI {@link ClientAuthenticationCache}
+     * The user is never anonymous since it must be authenticated to be stored in the cache
+     */
+    @Restricted(NoExternalUse.class)
+    public static class CLIUserDetails extends User {
```

Comment:
Why is this even `public`? It is used only from inside the class.

Anyway, it is created only from deprecated (Remoting-based) code, so it should also be `@Deprecated`, as a reminder to us to just delete all this stuff at some point.